flang/lib/Optimizer/Passes/Pipelines.cpp

```diff
@@ -264,10 +264,10 @@ void createDefaultFIRCodeGenPassPipeline(mlir::PassManager &pm,
   addNestedPassToAllTopLevelOperations(pm, fir::createAbstractResultOpt);
   fir::addCodeGenRewritePass(
       pm, (config.DebugInfo != llvm::codegenoptions::NoDebugInfo));
+  fir::createDebugPasses(pm, config.DebugInfo, config.OptLevel, inputFilename);
```

Comment:
Do you still get the correct assembly names for the external procedures (and common blocks) given `addExternalNameConversionPass` that goes from the `_QPfoo` to `foo_` name is now run afterwards?

You can move it before if needed, it does not need to be run after the target-rewrite and compiler-generated-name-conversion-pass.

Reply:
Thanks for catching this. I moved `ExternalNameConversion` early so that `linkageName` field in the debug is correct and added a test to check this.